Source/EasyNetQ/Consumer/ConsumerDispatcher.cs

```diff
@@ -34,25 +43,49 @@ public ConsumerDispatcher(ConnectionConfiguration configuration)
             thread.Start();
         }
 
-        public void QueueAction(Action action)
+        private bool IsDone()
+        {
+            return disposed && highPriority.IsEmpty && mediumPriority.IsEmpty && lowPriority.IsEmpty;
+        }
+
+        public void QueueAction(Action action, Priority priority = Priority.Low)
         {
             Preconditions.CheckNotNull(action, "action");
```

Comment:
```suggestion
            Preconditions.CheckNotNull(action, "action");
            if (disposed)
                throw new ObjectDisposedException(nameof(ConsumerDispatcher));
```

This check will be equivalent for `queue.CompleteAdding();` which disallow `queue.Add(action);` after being called.